internal/provisioner/equality/equality_test.go

```diff
@@ -531,6 +531,13 @@ func TestNodePortServiceChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if the ports number changed",
+			mutate: func(svc *corev1.Service) {
+				svc.Spec.Ports = svc.Spec.Ports[:len(svc.Spec.Ports)-1]
```

Comment:
also I think this test passes without any change to the implementation

you need to append to this list to get it to fail as you describe